File: Problems/0382-linked-list-random-node.py
import random

class Solution(object):

    def __init__(self, head):
        """
        :type head: Optional[ListNode]
        """
        self.head = head

    def getRandom(self):
        """
        :rtype: int
        """
        ans = self.head.val
        node = self.head.next
        i = 2

        while node:
            if random.randrange(i) == 0:
                ans = node.val
            node = node.next
            i += 1

        return ans

File: Problems/0382-linked-list-random-node.cpp
#include <bits/stdc++.h>
using namespace std;

struct ListNode {
    int val;
    ListNode* next;

    ListNode(int x) : val(x), next(nullptr) {}
};

// LeetCode solution starts from here
class Solution {
    ListNode* head;

public:
    Solution(ListNode* head) {
        this->head = head;
    }

    int getRandom() {
        int ans = head->val;
        ListNode* node = head->next;
        int i = 2;

        while (node) {
            if (rand() % i == 0)
                ans = node->val;

            node = node->next;
            i++;
        }

        return ans;
    }
};
